src/analysis/convex/basic.lean

```diff
@@ -557,9 +559,59 @@ begin
     exact hab }
 end
 
-variable {ι}
+@[nontriviality] lemma std_simplex_of_subsingleton [subsingleton 𝕜] : std_simplex 𝕜 ι = univ :=
+eq_univ_of_forall $ λ f, ⟨λ x, (subsingleton.elim _ _).le, subsingleton.elim _ _⟩
+
+/-- The standard simplex in the zero-dimensional space is empty. -/
+lemma std_simplex_of_empty [is_empty ι] [ne_zero (1 : 𝕜)] : std_simplex 𝕜 ι = ∅ :=
+eq_empty_of_forall_not_mem $ by { rintro f ⟨-, hf⟩, simpa using hf }
+
+lemma std_simplex_unique [unique ι] : std_simplex 𝕜 ι = {λ _, 1} :=
+begin
+  refine eq_singleton_iff_unique_mem.2 ⟨⟨λ _, zero_le_one, fintype.sum_unique _⟩, _⟩,
+  rintro f ⟨-, hf⟩,
+  rw [fintype.sum_unique] at hf,
+  exact funext (unique.forall_iff.2 hf)
+end
+
+variables {ι} [decidable_eq ι]
 
 lemma ite_eq_mem_std_simplex (i : ι) : (λ j, ite (i = j) (1:𝕜) 0) ∈ std_simplex 𝕜 ι :=
 ⟨λ j, by simp only; split_ifs; norm_num, by rw [finset.sum_ite_eq, if_pos (finset.mem_univ _)]⟩
 
+lemma single_mem_std_simplex (i : ι) : pi.single i 1 ∈ std_simplex 𝕜 ι :=
```

Comment:
I think this is missing a `DecidableEq`?

Reply:
There is a `decidable_eq` variable a few lines above.